File: app/test/e2e/channel_test.dart
import 'dart:convert';

import 'package:flutter_test/flutter_test.dart';
import 'package:argus/e2e/channel.dart';

void main() {
  test('plain channel round-trips params', () {
    final c = Channel.plain('c1');
    final params = utf8.encode('{"session_id":"s1"}');
    final frame = c.sealRequestFrame(1, 'sessions.tasks', 'node-a', params);
    final f = RelayFrame.parse(frame);
    expect(c.openParams(f), equals(params));
  });
}

File: internal/api/channel_test.go
package api

import (
	"encoding/json"
	"testing"
)

func TestPlainChannelRoundTrip(t *testing.T) {
	c := NewPlainChannel("c1")
	params := json.RawMessage(`{"session_id":"s1"}`)
	id := json.RawMessage(`1`)
	frame, err := c.SealRequestFrame(&id, "sessions.tasks", "node-a", params)
	if err != nil {
		t.Fatalf("seal: %v", err)
	}
	f, err := ParseRelayFrame(frame)
	if err != nil {
		t.Fatalf("parse: %v", err)
	}
	got, err := c.OpenParams(f)
	if err != nil {
		t.Fatalf("open: %v", err)
	}
	if string(got) != string(params) {
		t.Fatalf("round-trip mismatch: got %s want %s", got, params)
	}
}
